<h2> Exploratory Data Analysis Practice</h2>

<h4>Titanic: Machine Learning from Disaster <a href="https://www.kaggle.com/c/titanic/data">On Kaggle</a></h4>

The sinking of the Titanic is one of the most infamous shipwrecks in history.

On April 15, 1912, during her maiden voyage, the widely considered “unsinkable” RMS Titanic sank after colliding with an iceberg. Unfortunately, there weren’t enough lifeboats for everyone onboard, resulting in the death of 1502 out of 2224 passengers and crew.

While there was some element of luck involved in surviving, it seems some groups of people were more likely to survive than others.

In this challenge, we ask you to build a predictive model that answers the question: “what sorts of people were more likely to survive?” using passenger data (ie name, age, gender, socio-economic class, etc).

In [1]:
# 1. Importing libraries and viewing the different variables/columns

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

titanic = pd.read_csv("./titanic.csv")
titanic.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

<i><b>Observing above data, we can say passengerid, name , sex, ticket, cabin, embarked are general properties and they are non-numeric features so we can drop them.</b></i>

In [2]:
# Drop non-numeric or categorical features

drop_list = ["PassengerId", "Name", "Ticket", "Sex", "Cabin","Embarked"]
titanic.drop(drop_list, axis=1, inplace=True)
titanic.head()

Survived  Pclass   Age  SibSp  Parch     Fare
0         0       3  22.0      1      0   7.2500
1         1       1  38.0      1      0  71.2833
2         1       3  26.0      0      0   7.9250
3         1       1  35.0      1      0  53.1000
4         0       3  35.0      0      0   8.0500

<i><b>We have droped the unwanted features. Now we need to details about the availble numeric features.</b></i>

In [3]:
titanic.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

<i><b>We can observe from above data: 
    <ol>
        <li>Age have missing values, as count is only 714 whereas other features count is 891.</li>
        <li>As survived is binary, from mean we can say 38.38% people survived.</li>
        <li>Pclass is either 1, 2 or 3. Similary sibsp range from 0 to 8 and parch from 0 to 6.</li>
    </ol>
    Now we need to determine which features are strong indicator for survival i.e observe each feture and for that how many survived and how many didn't. 
</b></i>

In [4]:
# Group the data into two: survived and not survived. Find mean of all features for each group.
titanic.groupby('Survived').mean()

Pclass        Age     SibSp     Parch       Fare
Survived                                                    
0         2.531876  30.626179  0.553734  0.329690  22.117887
1         1.950292  28.343690  0.473684  0.464912  48.395408

<ul><li><i><b>A simple observation we can make from above is average age of survived people is 28.34 and not survived is 30.62. But the age have missing values so we need to observe it more.</b></i></li>
    <li><i><b>Fare and pclass between two groups stand out, so they might be useful </b></i></li></ul>

In [ ]:
# Check weather age is missing
titanic.groupby(titanic["Age"].isnull()).mean()